### ISSUE TITLE
Summary of daily data.

### ISSUE THREAD

**Issue by volloholic** [africanmathsinitiative/R-Instat#3516]
This dialogue should be easy to implement.  It should initially just need the standard summary code, with the factors being already completed.    So most of the complication will be to sort out the simplest possible dialogue.

**Comment by rdstern** [comment]
This dialogue should be easy to implement.  It can "borrow" structure and code from existing dialogues.  Nothing is needed that is not done (or soon will be) by the **Prepare > Column: Reshape > Column Statistics** dialogue.

It will just be made easier, because the columns being analysed, and the appropriate factor columns will be available automatically once the data frame has been defined to be climatic.

The left side of the dialogue can be as for the column summaries dialogue, down to the checkbox labelled **Print Results to Output**. 
Also, for now at least, include the following field, namely **Drop unused levels**. 
Also include the Summaries button, in the same position as in the Column Summaries dialogue.  And it includes the same summaries sub-dialogue.
The right-hand side will resemble the top of the Climatic > Prepare > Start of the Rains dialogue.

1. The first field is Station - as in Start of the rains!
2. The second is Date - as in Start of the rains
3. The third is Data - That's a different label.  But if it finds a Rain column (as in Start of the rains, then this is inserted automatically.
4. The 4th is Year - as in Start of the Rains
5. The fifth  has the label "Within Year".  If it finds a Month column then this is inserted there.

Test it initially with the Dodoma data - though that doesn't have a station variable.  Once James' Kenya data is there, then that is mainly rainfall.

The analyses can then be done with this dialogue and with the Column Statistics dialogue - with perhaps just one difference.  The simplest analyses will be without the Station or the Within Year field filled.  Then the summary is for the data element (usually rainfall) and the summary is by Year - as the factor.  
1) It should also work if there are no factors, i.e. even if the Year field is omitted.  So also could try (with multiple stations and no year.
2) The most complicated will be with Station, then Year, then Within Year.
3) There is one (minor) complication that is different to the Column Summaries dialogue.  The Year column may not be a factor column.  But here it has to be.  So the command will have to add an as.factor for this element.  (I hope - and assume it doesn't matter if that column is already a factor?)
4) I hope the summaries sub-dialogue will also be changing in parallel with this dialogue being constructed.  It will add more summaries and it will also become tabbed, with options for flexible handling of missing values.  That can wait here, until it is done generally.

**Comment by dannyparsons** [comment]
An initial implementation is now merged. I suggest leaving this open but moving to another milestone as it is not fully complete.

**Comment by dannyparsons** [comment]
duplicate of #2982
